freqtrade/exchange/__init__.py

```diff
@@ -48,390 +40,379 @@ def wrapper(*args, **kwargs):
-    except ccxt.BaseError as e:
-        raise OperationalException(e)
-
-
-@retrier
-def get_balance(currency: str) -> float:
-    if _CONF['dry_run']:
-        return 999.9
-
-    # ccxt exception is already handled by get_balances
-    balances = get_balances()
-    balance = balances.get(currency)
-    if balance is None:
-        raise TemporaryError(
-            f'Could not get {currency} balance due to malformed exchange response: {balances}')
-    return balance['free']
-
-
-@retrier
-def get_balances() -> dict:
-    if _CONF['dry_run']:
-        return {}
-
-    try:
-        balances = _API.fetch_balance()
-        # Remove additional info from ccxt results
-        balances.pop("info", None)
-        balances.pop("free", None)
-        balances.pop("total", None)
-        balances.pop("used", None)
-
-        return balances
-    except (ccxt.NetworkError, ccxt.ExchangeError) as e:
-        raise TemporaryError(
-            f'Could not get balance due to {e.__class__.__name__}. Message: {e}')
-    except ccxt.BaseError as e:
-        raise OperationalException(e)
-
-
-@retrier
-def get_tickers() -> Dict:
-    try:
-        return _API.fetch_tickers()
-    except ccxt.NotSupported as e:
-        raise OperationalException(
-            f'Exchange {_API.name} does not support fetching tickers in batch.'
-            f'Message: {e}')
-    except (ccxt.NetworkError, ccxt.ExchangeError) as e:
-        raise TemporaryError(
-            f'Could not load tickers due to {e.__class__.__name__}. Message: {e}')
-    except ccxt.BaseError as e:
-        raise OperationalException(e)
-
-
-@retrier
-def get_ticker(pair: str, refresh: Optional[bool] = True) -> dict:
-    global _CACHED_TICKER
-    if refresh or pair not in _CACHED_TICKER.keys():
+class Exchange(object):
+
+    # Current selected exchange
+    _API: ccxt.Exchange = None
+    _CONF: Dict = {}
+    _CACHED_TICKER: Dict[str, Any] = {}
+
+    # Holds all open sell orders for dry_run
+    _DRY_RUN_OPEN_ORDERS: Dict[str, Any] = {}
+
+    def __init__(self, config: dict) -> None:
+        """
+        Initializes this module with the given config,
+        it does basic validation whether the specified
+        exchange and pairs are valid.
+        :param config: config to use
+        :return: None
+        """
+        self._API
+
+        self._CONF.update(config)
+
+        if config['dry_run']:
+            logger.info('Instance is running with dry_run enabled')
+
+        exchange_config = config['exchange']
+        self._API = self._init_ccxt(exchange_config)
+
+        logger.info('Using Exchange "%s"', self.get_name())
+
+        # Check if all pairs are available
+        self.validate_pairs(config['exchange']['pair_whitelist'])
+
+    def _init_ccxt(self, exchange_config: dict) -> ccxt.Exchange:
+        """
+        Initialize ccxt with given config and return valid
+        ccxt instance.
+        :param config: config to use
```

Comment:
the is no longer a `config` parameter: IMO we can drop such comments (especially for private methods) as the typehint already does a pretty good job describing it.